src/main/java/com/firebolt/jdbc/service/FireboltEngineInformationSchemaService.java

```diff
@@ -66,12 +81,13 @@ private Engine getEngine(String engine, @Nullable String database) throws SQLExc
                 if (database != null && !database.equals(attachedDatabase)) {
                     throw new FireboltException(format("The engine with the name %s is not attached to database %s", engine, database));
                 }
-                return new Engine(rs.getString(ENGINE_URL), status, rs.getString(ENGINE_NAME_FIELD), attachedDatabase, null);
+                String url = rs.getString(ENGINE_URL).split("\\?", 2)[0];
```

Comment:
In https://github.com/firebolt-db/jdbc/pull/311 i was grabbing the query params because we need to set those to get correct routing to the engine if there are any present.

Reply:
Done.